hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/service/ContractCallService.java

```diff
@@ -152,6 +157,23 @@ private void restoreGasToBucket(HederaEvmTransactionProcessingResult result, lon
         }
     }
 
+    protected void validateResult(
+            final HederaEvmTransactionProcessingResult txnResult, final CallServiceParameters params) {
+        if (!txnResult.isSuccessful()) {
+            updateGasUsedMetric(ERROR, txnResult.getGasUsed(), 1);
+            var revertReason = txnResult.getRevertReason().orElse(Bytes.EMPTY);
+            var detail = maybeDecodeSolidityErrorStringToReadableMessage(revertReason);
+            throw new MirrorEvmTransactionException(
+                    getStatusOrDefault(txnResult).name(),
+                    detail,
+                    revertReason.toHexString(),
+                    txnResult,
+                    mirrorNodeEvmProperties.isModularizedServices() && params.isModularized());
```

Comment:
Isn't it redundant to check the properties since the param can only be true if the prop is also true?
```suggestion
                    params.isModularized());
```

Reply:
The params isModularized currently is only calculated based on the percentage. I think we can improve that in a follow up PR to also check mirrorNodeEvmProrties in directTrafficThroughTransactionExecutionService. 
Current logic is: 
```
boolean isModularized = evmProperties.directTrafficThroughTransactionExecutionService();
public boolean directTrafficThroughTransactionExecutionService() {
        return RandomUtils.secure().randomDouble(0.0d, 1.0d) < getModularizedTrafficPercent();
}
```

Reply:
Added some tests for that in https://github.com/hiero-ledger/hiero-mirror-node/pull/10619

Reply:
This should still work. Can get this pr in with priority and will consolidate any other changes in the other PR.

Reply:
Merged the other PR first since this PR has test failures.